sympy/sets/sets.py

```diff
@@ -867,7 +867,7 @@ def __len__(self):
         return reduce(lambda a, b: a*b, (len(s) for s in self.args))
 
     def __bool__(self):
-        return all([bool(s) for s in self.sets])
+        return all(bool(s) for s in self.sets)
```

Comment:
Is `bool` needed inside `all`?

Reply:
I would have thought this could just be `all(self.sets)`.

Reply:
Thanks! Changed it and it seems to work.